### ISSUE TITLE
Cannot pass record key through CYPRESS_RECORD_KEY variable (local dev mode)

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#820]
### bug

When working inside this repository and trying to record a run, the record key is not read from the environment variable `CYPRESS_RECORD_KEY`.

### steps to reproduce

1. clone a sample project for example `git clone https://github.com/cypress-io/cypress-test-tiny.git /tmp/cypress-test-tiny`
2. try running tests in the `/tmp/test` from the root of this repository AND record it (because we are working against local API without running it, just use fake "foo" and "bar").

```
CYPRESS_PROJECT_ID=foo npm start -- --run-project /tmp/cypress-test-tiny --record --key bar
```
Everything is working and the test is running.

3. try passing record key via environment variable `CYPRESS_RECORD_KEY` (see https://docs.cypress.io/dashboard/overview/projects-dashboard.html#Record-Test-Runs) 

```
CYPRESS_PROJECT_ID=foo CYPRESS_RECORD_KEY=bar DEBUG=cypress:* npm start -- --run-project /tmp/cypress-test-tiny --record
```

and you get the following error message

```
You passed the --record flag but did not provide us your Record Key.

You can pass us your Record Key like this:

  cypress run --record --key <record_key>

You can also set the key as an environment variable with the name CYPRESS_RECORD_KEY.

https://on.cypress.io/how-do-i-record-runs
```

Hmm, weird that this fails in local development mode, but works just fine in compiled binary.

**Comment by bahmutov** [comment]
Hmm, in local dev mode when using `start` we are going through [scripts/start.js](scripts/start.js) which is very different from CLI logic

**Comment by brian-mann** [comment]
Fixed in [`1.0.3`](https://docs.cypress.io/guides/references/changelog.html#1-0-3).
